### ISSUE TITLE
1-letter typo in the definition of scipy.special.spence function in the documentation

### ISSUE THREAD

**Issue by jjhjm17** [scipy/scipy#13628]
My issue is about integration range for the definition of scipy.special.spence function in the documentation.

#### Reproducing code example:

```
from scipy.special import spence
spence(0)  # 1.6449340668482264
spence(1)  # 0.0
import numpy as np
np.pi**2/6 - spence(0)  # 0.0
```

#### Error message:

There is no error message. However, according to the documentation,
spence(z) is defined as Int_0 ^z  log(t)/(1-t) dt.
So spence(0) is integration from 0 to 0 and it should be 0.
However, it is pi^2/6. This value is consistent to the second definition in the notes section in the documentation.
In the notes section, spence(1-z) = Int_0 ^z log(1-t)/t dt. 
When z = 1,  spence(1-1) = spence(0) = Int_0 ^1 log(1-t)/t dt = pi^2/6.
Please see the wikipedia's spence function (https://en.wikipedia.org/wiki/Spence%27s_function), 3rd equation and special values section.

In my opinion, at the first definition of documentation,
the integration range should not be 0 to z, but from 1 to z. This is 1-letter typo in my opinion. Then it is consistent with the second definition in the notes section and consistent with the current implementation.

Thank you very much!

#### Scipy/Numpy/Python version information:
 1.6.1 1.19.2 sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)

**Comment by jjhjm17** [comment]
A short summary:
scipy.special.spence(0) should be 0, because it is integration from 0 to 0, according to its definition in the documentation. However, it is nonzero. My suggestion is to change the integration range not from 0, but from 1 to z, in the definition of documentation. This definition is consistent to the value spence(0)=pi^2/6 and to the other definition in the notes section in the documentation. Thank you very much!

**Comment by rajpratyush** [comment]
@ev-br Can i take up this issue?

**Comment by ev-br** [comment]
Sure. You not need permission though, just go ahead and work on this or any other issue. 

For this one however, the valuable part is checking consistency between the actual imlementation and the docs.

**Comment by rajpratyush** [comment]
@ev-br since it is my first time contributing here could you point me to the file need to be changed

**Comment by mdhaber** [comment]
Good first issue. Not sure where it's defined; I'd need to do some digging myself. Consider searching the text of files in the `scipy/scipy/special` directory.

**Comment by ilayn** [comment]
The relevant part of the docs is here 

https://github.com/scipy/scipy/blob/9492369e7f76c8206ac1c375e75375ca7410a7e7/scipy/special/add_newdocs.py#L8884-L8919

**Comment by shashankks0987** [comment]
test_solve_generalized_discrete_are , TestPareto.test_fit_MLE_comp_optimzer[5-2-1] I have these two tests failing after build, any idea what to do?

**Comment by tylerjereddy** [comment]
Fixed by #14060
